.NET 8 support

I'm currently using .NET 8 RC2. When I try to install csharpier in my CI/CD pipeline on Azure Pipelines, I get the following error:

```
- script: |
    dotnet tool restore
    dotnet csharpier --check .
  displayName: "Lint Pull Request"
```

```
You must install or update .NET to run this application.

App: /home/vsts/.nuget/packages/csharpier/0.25.0/tools/net7.0/any/dotnet-csharpier.dll
Architecture: x64
Framework: 'Microsoft.NETCore.App', version '7.0.0' (x64)
.NET location: /opt/hostedtoolcache/dotnet/

The following frameworks were found:
  8.0.0-rc.2.23479.6 at [/opt/hostedtoolcache/dotnet/shared/Microsoft.NETCore.App]
```

Could support for RC2 be added? Or perhaps support for .NET 8 as soon as it is out?

@belav  I do think some changes are needed for C# 12 support.

I have a file that contains the following code:

```
        Func<Exception, IActionResult>[] handlers =
        [
            // You can add handlers for specific exceptions here.
            // Examples: Domain validation, unauthorized exceptions, etc...
        ] ;
```

This ` = []` is a new C# 12 concept. When I run `dotnet csharpier .`, I get the following error:

```
Warning <STUFF>/ErrorController.cs - Failed to compile so was not formatted.
  (27,52): error CS1525: Invalid expression term '['
  (28,10): error CS0443: Syntax error; value expected
```

--- a/.config/dotnet-tools.json
+++ b/.config/dotnet-tools.json
@@ -9,7 +9,7 @@
       ]
     },
     "husky": {
-      "version": "0.5.4",
+      "version": "0.6.2",
       "commands": [
         "husky"
       ]

--- a/Directory.Build.props
+++ b/Directory.Build.props
@@ -1,6 +1,6 @@
 <Project>
   <PropertyGroup>
-    <TargetFrameworks>net7.0</TargetFrameworks>
+    <TargetFrameworks>net8.0</TargetFrameworks>
     <LangVersion>11</LangVersion>
     <ImplicitUsings>enable</ImplicitUsings>
     <Nullable>enable</Nullable>

--- a/Shell/BuildProject.psm1
+++ b/Shell/BuildProject.psm1
@@ -2,7 +2,6 @@ function CSH-BuildProject
 {
     $repositoryRoot = Join-Path $PSScriptRoot ".."
     $csProjectPath = Join-Path $repositoryRoot "Src/CSharpier.Cli/CSharpier.Cli.csproj"
-    $csharpierDllPath = Join-Path $repositoryRoot "Src/CSharpier.Cli/bin/release/net7.0/dotnet-csharpier.dll"
     
     & dotnet build $csProjectPath -c release
     if ($lastExitCode -gt 0) {

--- a/Shell/ReviewBranch.psm1
+++ b/Shell/ReviewBranch.psm1
@@ -7,7 +7,7 @@ function CSH-ReviewBranch {
 
     $repositoryRoot = Join-Path $PSScriptRoot ".."
     $csProjectPath = Join-Path $repositoryRoot "Src/CSharpier.Cli/CSharpier.Cli.csproj"
-    $csharpierDllPath = Join-Path $repositoryRoot "Src/CSharpier.Cli/bin/release/net7.0/dotnet-csharpier.dll"
+    $csharpierDllPath = Join-Path $repositoryRoot "Src/CSharpier.Cli/bin/release/net8.0/dotnet-csharpier.dll"
 
     $location = Get-Location
 

--- a/Src/CSharpier.Cli/CSharpier.Cli.csproj
+++ b/Src/CSharpier.Cli/CSharpier.Cli.csproj
@@ -4,7 +4,7 @@
     <OutputType>Exe</OutputType>
     <PackageId>CSharpier</PackageId>
     <AssemblyName>dotnet-csharpier</AssemblyName>
-    <TargetFrameworks>net6.0;net7.0</TargetFrameworks>
+    <TargetFrameworks>net6.0;net7.0;net8.0</TargetFrameworks>
     <PackAsTool>true</PackAsTool>
     <AssemblyOriginatorKeyFile>../../Nuget/csharpier.snk</AssemblyOriginatorKeyFile>
     <SignAssembly>True</SignAssembly>

--- a/Src/CSharpier.Generators/CSharpier.Generators.csproj
+++ b/Src/CSharpier.Generators/CSharpier.Generators.csproj
@@ -1,6 +1,6 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>net6.0;net7.0;netstandard2.0</TargetFrameworks>
+    <TargetFrameworks>net6.0;net7.0;net8.0;netstandard2.0</TargetFrameworks>
     <EmitCompilerGeneratedFiles>true</EmitCompilerGeneratedFiles>
     <EnforceExtendedAnalyzerRules>true</EnforceExtendedAnalyzerRules>
     <PackageScribanIncludeSource>true</PackageScribanIncludeSource>

--- a/Src/CSharpier.MsBuild/CSharpier.MsBuild.csproj
+++ b/Src/CSharpier.MsBuild/CSharpier.MsBuild.csproj
@@ -1,8 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
     <Import Project="../../Nuget/Build.props"/>
     <PropertyGroup>
-        <!-- See https://github.com/belav/csharpier/issues/481 and https://github.com/belav/csharpier/issues/773 -->
-        <TargetFrameworks>net6.0;net7.0</TargetFrameworks>
+        <TargetFrameworks>net6.0;net7.0;net8.0</TargetFrameworks>
         <PackageId>CSharpier.MsBuild</PackageId>
         <DevelopmentDependency>true</DevelopmentDependency>
         <CSharpierOutputDir>../CSharpier.Cli/bin/$(Configuration)/$(TargetFramework)</CSharpierOutputDir>

--- a/Src/CSharpier/CSharpier.csproj
+++ b/Src/CSharpier/CSharpier.csproj
@@ -2,7 +2,7 @@
   <Import Project="../../Nuget/Build.props" />
   <PropertyGroup>
     <PackageId>CSharpier.Core</PackageId>
-    <TargetFrameworks>net6.0;net7.0;netstandard2.0</TargetFrameworks>
+    <TargetFrameworks>net6.0;net7.0;net8.0;netstandard2.0</TargetFrameworks>
     <RootNamespace>CSharpier</RootNamespace>
     <EmitCompilerGeneratedFiles>true</EmitCompilerGeneratedFiles>
     <AssemblyOriginatorKeyFile>../../Nuget/csharpier.snk</AssemblyOriginatorKeyFile>

--- a/global.json
+++ b/global.json
@@ -1,6 +1,6 @@
 {
   "sdk": {
-    "version": "7.0.400",
+    "version": "8.0.100-rc.2.23502.2",
     "rollForward": "latestFeature"
   }
 }

